### PR TITLE
Implement checksum write for `krel push`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This adds the checksum write to `krel push` by introducing a new API `release.WriteChecksums()`.
#### Which issue(s) this PR fixes:

Refers to #1459 
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `release.WriteChecksums()` API for writing release checksums (mainly used by `krel push`)
```

### CODE REVIEW

File: pkg/release/release.go [hunk 1]
```diff
@@ -360,3 +365,102 @@ func CopyBinaries(rootPath string) error {
...
+				"xargs", "-0", "-I", "{}", "shasum", "-a",
+				digest, filepath.Join(rootPath, "{}"),
+			).
+			AddOutputWriter(file).
+			RunSuccess(); err != nil {
+			return "", errors.Wrap(err, "create SHA sum file contents")
+		}
+
+		return file.Name(), nil
+	}
```
Comment (hasheddan) [inline pkg/release/release.go:394]: Any thoughts on using the `crypto` pkg for this rather than shelling out command?

Comment (saschagrunert) [reply pkg/release/release.go]: Done :+1:

**Pull request merged**